server/server.py

```diff
@@ -48,15 +51,35 @@ def handle(self):
 
 def keyGen(path):
 	key = RSA.generate(2048)
-	with open(path +'/python.pem','w') as privateKey:
+	with open(path +'/python.pem','wb') as privateKey:
 		privateKey.write(key.exportKey('PEM'))
-	with open(path+ '/python.pub', 'w') as publicKey:
-		publicKey.write(key.publickey.exportKey('PEM'))
+	with open(path+ '/python.pub', 'wb') as publicKey:
+		publicKey.write(key.publickey().exportKey('PEM'))
+
+def parseArgs():
+	'''Parses args using the argparse lib'''
+	parser = argparse.ArgumentParser(description='Location logging server')
+
+	parser.add_argument('-g', '--generate-keys', metavar='PATH', type=str)
```

Comment:
Ah, indeed. I just saw this in the [argparse documentation](https://docs.python.org/3/library/argparse.html#nargs): 

> Note that nargs=1 produces a list of one item. This is different from the default, in which the item is produced by itself.